src/expr/src/expr/expr_regexp.rs

```diff
@@ -248,3 +252,513 @@ impl Expression for RegexpMatchExpression {
+                        let mut search_start = start as usize;
+                        while let Some(capture) = self.ctx.0.captures(&text[search_start..]) {
+                            // Get the current start & end index
+                            let match_start = capture.get(0).unwrap().start();
+                            let match_end = capture.get(0).unwrap().end();
+
+                            if count == self.n {
+                                // We've reached the pattern to replace
+                                // Let's construct the return string
+                                ret = format!(
+                                    "{}{}{}",
+                                    &text[..search_start + match_start],
+                                    self.replacement.clone(),
+                                    &text[search_start + match_end..]
+                                );
+                                break;
+                            }
+
+                            // Update the counter
+                            count += 1;
+
+                            // Update `start`
+                            search_start += match_end;
+                        }
+                    }
+                } else {
+                    // There are capture groups in the regex
+                    println!("Path Four");
+                    // Reset return string at the beginning
+                    ret = "".to_string();
+                    if self.n == -1 {
+                        // `N` is not specified
+                        if let Some(capture) = self.ctx.0.captures(&text[start as usize..]) {
+                            let match_start = capture.get(0).unwrap().start();
+                            let match_end = capture.get(0).unwrap().end();
+
+                            // Get the replaced string and expand it
+                            let replaced = self
+                                .regex
+                                .replace_all(&self.replacement, "$${$1}")
+                                .to_string();
+                            let mut expanded = String::new();
+                            capture.expand(&replaced, &mut expanded);
+
+                            // Construct the return string
+                            ret = format!(
+                                "{}{}{}",
+                                &text[..start as usize + match_start],
+                                expanded,
+                                &text[start as usize + match_end..]
+                            );
+                        } else {
+                            // No match
+                            ret = text.into();
+                        }
+                    } else {
+                        // Replace only the N-th match
+                        let mut count = 1;
+                        while let Some(capture) = self.ctx.0.captures(&text[start as usize..]) {
```

Comment:
Feels like this branch can be merged with one of the previous branch.

Reply:
This could be merged, but it will let the overall logic and data flow become ambiguous I think🤔️
Since the difference between this branch and the above one is about the capture group and the use of `expand`.

Reply:
No problem. It's fine as long as the logic is clear.